datafusion/sql/src/expr/value.rs

```diff
@@ -169,7 +169,7 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
             }
         }
 
-        internal_err!("Expected a simplified result, but none was found")
+        not_impl_err!("Could not plan array literal. Hint: is `nested_expressions` DataFusion feature selected?")
```

Comment:
I would say this question it is not clear a little bit. `nested_expressions` DataFusion feature selected - is it cause of issue or solution? 

```suggestion
        not_impl_err!("Could not plan array literal. Hint: Please try with `nested_expressions` DataFusion feature enabled")
```

Reply:
That is a good call -- thank you @comphead